### DISCUSSION TITLE
应用商店安装halo2.15.0使用MariaDB数据库自动生成的参数错误，导致容器一直重启

### DISCUSSION THREAD

**Discussion by sofeiall** [1Panel-dev/1Panel#5059] [Q&A]
### 联系方式

hnlink@qq.com

### 1Panel 版本

V1.10.6-lts

### 问题描述

应用商店安装halo2.15.0使用MariaDB数据库自动生成的参数错误，导致容器一直重启[BUG]

应用商店安装 自动生成的参数 如下：
'--spring.r2dbc.url=r2dbc:pool:**mariadb**://mariadb:3306/halo_s6pxjf' '--spring.r2dbc.username=halo_hs8sMP' '--spring.r2dbc.password=halo_nBTSRJ' '--spring.sql.init.platform=mariadb' '--halo.external-url=http://localhost:8080'

导致无法连接MariaDB数据库，容器一直重启。

手动修改参数把： r2dbc:pool:**mariadb**，修改为：r2dbc:pool:**mysql** 可以正常连接数据库，网站正常。

'--spring.r2dbc.url=r2dbc:pool:**mysql**://mariadb:3306/halo_s6pxjf' '--spring.r2dbc.username=halo_hs8sMP' '--spring.r2dbc.password=halo_nBTSRJ' '--spring.sql.init.platform=mariadb' '--halo.external-url=http://localhost:8080'

### 重现步骤

应用商店安装halo应用，使用MariaDB数据库

**Comment by wanghe-fit2cloud** [comment]
目前 Halo 官方给出的修复建议是修改数据库链接地址，临时采用 mysql 驱动。
因此，在安装 Halo 选择 MariaDB 数据库时，可以先手动修改 compose 来支持上述方案，如图：
![image](https://github.com/1Panel-dev/1Panel/assets/46734380/faad1acb-759f-4356-b178-408116f21d92)

> 我们近期也会跟踪一下上述问题。
